### PR TITLE
Support getting Git Parameter choices via extra call 

### PR DESCRIPTION
**Pull request by chrisxiao**
I have the same problem in #258. 
I  found the url
$JOB_URL/descriptorByName/net.uaznia.lukanus.hudson.plugins.gitparameter.GitParameterDefinition/fillValueItems?param=$PARAM_NAME 
can return the params choices values, so fix it.

### CODE REVIEW

**Comment by MCMicS** [comment]
Thanks
i will have a look after holiday

File: src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java [hunk 1]
```diff
@@ -117,7 +118,20 @@ private void addParameterInputs() {
         for (JobParameter jobParameter : parameters) {
             final JobParameterRenderer jobParameterRenderer = JobParameterRenderer.findRenderer(jobParameter)
                     .orElseGet(ErrorRenderer::new);
-            final JobParameterComponent<?> jobParameterComponent = jobParameterRenderer.render(jobParameter);
+            JobParameterComponent<?> jobParameterComponent;
+            if (jobParameter.getChoices().isEmpty() && jobParameterRenderer instanceof GitParameterRenderer) {
```
Comment (MCMicS) [inline src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java:122]: Should this logioc not be added in `GitParameterRenderer`
e.g load values if not available

Comment (chrisxiao) [reply src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java:122]: I checked the code of `GitParameterRenderer `, how about adding this to its `render` method ?

Comment (MCMicS) [reply src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java:122]: You can retreive a service for RequestManager and then load need ed data.

I try to edit your branch the days.

Comment (chrisxiao) [reply src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java:122]: ok.

Comment (MCMicS) [reply src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java]: I changed the render pipeline to allow loading of additional Values for each renderer.
Could you please check if it still works for you

I also improved it with URI Encoding an dadd support an additional Git Parameter Plugin (ListGitBranchesParameter)

**Review by MCMicS — APPROVED** [review]

File: src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java (hunk 1 above)
Comment (chrisxiao) [reply src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java]: I tested, it's ok. Thanks

**Pull request merged**